Fabric.Authorization.AccessControl/src/app/modules/access-control/member-add/member-add.component.ts

```diff
@@ -7,9 +17,76 @@ import { Component, OnInit } from '@angular/core';
 })
 export class MemberAddComponent implements OnInit {
 
-  constructor() { }
+  private grain: string = 'app';
+  public searchInput: string;
+  public principals: Array<FabricPrincipal>;
+  public selectedPrincipal: FabricPrincipal;
+  public roles: Array<Role>;
+  public selectedRoles: Array<Role>;  
+  
+
+  constructor(private idpSearchService: FabricExternalIdpSearchService, 
+    private roleService: FabricAuthRoleService,
+    private userService: FabricAuthUserService,
+    private configService: AccessControlConfigService,
+    private groupService: FabricAuthGroupService
+    ) {
+      this.selectedRoles = [];
+     }
 
   ngOnInit() {
+    this.getRoles();
+  }
+
+  onKey(searchText){
+    if(searchText.length < 2) {
+      return;
+    }
+    var request = new IdPSearchRequest();
+    request.searchText = searchText;
+
+    this.idpSearchService.searchExternalIdP(request)
+    .subscribe(result => {
+      this.principals = result.principals;
+    });
   }
 
+  onPrincipalSelect(principal: FabricPrincipal){
+    this.selectedPrincipal = principal;
+  }
+
+  onRoleSelect(role: Role){
+    if(!this.roleIsSelected(role)){
+      this.selectedRoles.push(role);
+    } else{
+      this.selectedRoles = this.selectedRoles.filter(r => r.name !== role.name);
+    }
+  }
+
+  roleIsSelected(role: Role){
+    return this.selectedRoles.filter(r => r.name === role.name).length > 0;
+  }
+
+  getRoles(){
```

Comment:
This is passing in a `clientId` and not a `securableItem` for the second parameter.

Reply:
ahh yeah. they are the same value but i will switch it. a lot of this code is probably going away anyway once the data governance team does the real ui